src/NuGet.Services.EndToEnd/Support/Clients/V3IndexClient.cs

```diff
@@ -104,10 +105,16 @@ private static List<string> GetResourceUrls(V3Index v3Index, Func<Resource, bool
 
         private async Task<V3Index> GetV3IndexAsync(ITestOutputHelper logger)
         {
-            return await _httpClient.GetJsonAsync<V3Index>(
-                _testSettings.V3IndexUrl,
-                allowNotFound: false,
-                logResponseBody: false,
+            return await RetryUtility.ExecuteWithRetry(
+                async () =>
+                {
+                    return await _httpClient.GetJsonAsync<V3Index>(
+                        _testSettings.V3IndexUrl,
+                        allowNotFound: false,
+                        logResponseBody: false,
+                        logger: logger);
+                },
+                ex => (ex.HasTypeOrInnerType<WebException>(out var we) && we.Status == WebExceptionStatus.NameResolutionFailure),
```

Comment:
How often did it happen? Is it a concern?

Reply:
Regarding, "Is it a concern?" generally speaking yes. Any E2E failure indicates a potential product issue. That being said, I looked at availability tests and CatchPoint tests and saw no similar symptom in the same period as the failures. Also, I don't recall any user reports about this problem, ever.

This suggests to me that it's related to our test agents and not a real customer impacting issue. IMO, we can depend on monitoring for this sort of failure and this change will simply help our deployment agility.